platform/linux-generic/pktio/socket_xdp.c

```diff
@@ -450,11 +447,21 @@ static uint32_t process_received(pktio_entry_t *pktio_entry, xdp_sock_info_t *so
 				continue;
 			}
 
-			if (pktio_cls_enabled(pktio_entry) &&
-			    _odp_cls_classify_packet(pktio_entry, pkt_data.data, pool_hdl,
-						     pkt_data.pkt_hdr)) {
-				odp_packet_free(pkt_data.pkt);
-				continue;
+			if (pktio_cls_enabled(pktio_entry)) {
+				odp_pool_t new_pool;
+
+				ret = _odp_cls_classify_packet(pktio_entry, pkt_data.data,
+							       &new_pool, pkt_data.pkt_hdr);
+				if (ret) {
+					odp_packet_free(pkt_data.pkt);
+					continue;
+				}
+
+				if (odp_unlikely(_odp_pktio_packet_to_pool(
```

Comment:
This probably needs to be done after `pkt_hdr->seg_data` and `pkt_hdr->event_hdr.base_data` are set (below), otherwise the new packet will have data that points to the old pool.